### DISCUSSION TITLE
please include a way to extract audio files and video files

### DISCUSSION THREAD

**Discussion by rgb-git** [ShapeCrawler/ShapeCrawler#265] [Ideas]
BinaryData is always null for audio and video shape types.

**Comment by AdamShakhabov** [comment]
@rgb-git, the bug has been fixed in [**v0.31.2**](https://github.com/ShapeCrawler/ShapeCrawler/releases/tag/v0.31.2)

**Reply by rgb-git** [reply]
Thank you so much, this will help me with my project greatly !

**Comment by rgb-git** [comment]
I have downloaded the update and I am able to extract the audio and video successfully.
I am sorry for not mentioning this earlier but could you expose the MIME type for IAudioShape and IVideoShape (similar to the IPicture)?
I need to save the audio and video data as files and I need to know what file extension to use.
If there is already a way to get the file extension, please let me know.
Thanks again!

**Reply by AdamShakhabov** [reply]
Ok. Will take a look at this feature.

**Reply by AdamShakhabov** [reply]
@rgb-git, reading MIME type has been added in **[v0.32.0](https://github.com/ShapeCrawler/ShapeCrawler/releases/tag/v0.32.0)**

**Reply by rgb-git** [reply]
Thank you for adding the MIME type to the audio and video shapes.
When I tested this using a powerpoint file that contains one "m4a" audio file, the MIME that got returned was "mp4".
(I unzipped the powerpoint file and verified that the embedded audio file is named media1.m4a.)
Would it be possible to return the actual file extension (m4a) for the MIME type?
If not, is there a way to add a new property to return the name of the embedded audio file (media1.m4a)?
Thanks again for your attention to this !

**Reply by AdamShakhabov** [reply]
My quick research says that `audio/mp4` is valid MIME for `.m4a`. [There is no internal difference between](https://fileinfo.com/extension/m4a) `.mp4` and `.m4a` files.

As you suggested, we will add a new property to return the name of the embedded audio file.

**Reply by AdamShakhabov** [reply]
@rgb-git, please check [**v0.34.0**](https://github.com/ShapeCrawler/ShapeCrawler/releases/tag/v0.34.0) version. Now you can get file name via `IImage.Name` property.

**Reply by rgb-git** [reply]
Awesome, thank you
